<launch>
  <arg name="use_yaml" default="true" />
  <arg name="yaml_file" default="" />
  <arg name="recieving_mode" default="url" doc="Google Chat API settings. Choose from url, dialogflow."/>
  <arg name="download_data" default="true" />
  <arg name="download_data_timeout" default="10" />
  <arg name="download_avator" default="false" />
  <arg name="google_cloud_credentials_json" />
  <arg name="host" default=""/>
  <arg name="port" default=""/>
  <arg name="ssl_certfile" default=""/>
  <arg name="ssl_keyfile" default=""/>
  <arg name="respawn" default="true" />

  <node name="google_chat_ros" pkg="google_chat_ros" type="google_chat_ros_server.py"
        respawn="$(arg respawn)" output="screen">
    <rosparam command="load" file="$(arg yaml_file)"
              if="$(arg use_yaml)" />
    <rosparam subst_value="true">
      recieving_mode: $(arg recieving_mode)
      download_timeout: $(arg download_timeout)
      download_data: $(arg download_data)
      download_avator: $(arg download_avator)
      google_cloud_credentials_json: $(arg google_cloud_credentials_json)
      host: $(arg host)
      port: $(arg port)
      ssl_certfile: $(arg ssl_certfile)
      ssl_keyfile: $(arg ssl_keyfile)
    </rosparam>
  </node>
</launch>
